### DISCUSSION TITLE
🚀 Projects is now generally available!

### DISCUSSION THREAD

**Discussion by 2percentsilk** [community/community#27327] [Projects and Issues]
After exactly nine months in Public Preview and [15 changelogs](https://github.blog/changelog/label/issues/) of updates, we're excited to announce that Projects is generally available for existing Free, Team, and Enterprise (Cloud) customers 🥳

Check out the [blog](https://github.blog/2022-07-27-planning-next-to-your-code-github-projects-is-now-generally-available) for more details about our journey and what's coming next ⏩

💖 Thank you all 💖 for your continued feedback during our public beta - you've helped us learn fast and continue to improve every part of the product! We truly wouldn't be here without you and we look forward to continuing this journey to a more delightful experience for planning and tracking your work!

**Comment by super-cooper** [comment]
Congratulations! I would like to migrate my old projects to the new platform, but I don't see a button to do this. Is it possible to migrate "classic" projects? I recall this being possible during the beta.

**Reply by shiftkey** [reply]
If you open the sidebar there's a menu with a **Migrate** menu item which will walk you through the process:

<img width="362" src="https://user-images.githubusercontent.com/359239/181349968-1cceadff-2fa5-4b62-8a16-dcd21996fdae.png">

**Reply by super-cooper** [reply]
@shiftkey Thanks for your reply! I recalled seeing this menu option during the beta, but it doesn't appear to be there for me:
![image](https://user-images.githubusercontent.com/14908880/181353845-f54b5384-a287-4966-869f-9e4a54ec89bb.png)

Does it only work for certain types of projects? I have a repository with 8 projects and none of them are showing the option in the menu.

**Reply by shiftkey** [reply]
@super-cooper thanks for that screenshot. We'll investigate things on our end and let you know if we need any more information about the project.

**Reply by super-cooper** [reply]
Thank you, I appreciate that :) Would be happy to provide more information if you need it.

**Reply by 2percentsilk** [reply]
@super-cooper, you'll need to enable the migration feature (it's currently in feature preview!)

More detailed instructions [in our migration docs](https://docs.github.com/en/issues/planning-and-tracking-with-projects/creating-projects/migrating-from-projects-classic).

Let us know if that helps - @shiftkey and I have been testing it so easy to forget you need to turn it on 😅

**Reply by super-cooper** [reply]
No worries, I have done similar things many times :) Thank you for your help

**Reply by Gitleyla** [reply]
Ich brauche Hilfe bitte bitte 
Bin noch betrogen worden Um 90 tausend€
Wer ist so nett bin verzweifelt 
TEL 01726648770 
ersinoba693@gmail.com
Bitte ihr Lieben hilft mir .

**Reply by atirutw** [reply]
Oh god they're here too

**Comment by eragon11** [comment]
Awesome, it's really great updates.

**Reply by Aankk98** [reply]
ALyciél D'âmour

**Comment by Gani-coder** [comment]
Hello i need help from you, how much i have to wait for DNS check in progres , i host e record from Github to Bluehos at TXT and limited time to BlueHost it was 4h (min) does i have to wait until DNS check from BlueHost or what to do please HELP ME THANKYOU.

**Comment by leonardowf** [comment]
Congratulations to the team! It looks awesome! 💯

**Comment by ardrigh** [comment]
Please add a default workflow for adding Pull Requests separate from adding Issues!!

When we add Pull Requests to our board we want them to go to a "Code Review" status and Issues go to a "Backlog" status.

We do not act like Issues and Pull Request are the same thing.

**Reply by 2percentsilk** [reply]
Thanks for the feedback! We are investigating new workflows/automations to add to the product and will keep this in mind for as we plan our next features.

**Comment by mchelen-gov** [comment]
This is super exciting! It seems that Github Mobile app cannot view Project boards though, is this correct?

**Reply by 2percentsilk** [reply]
I believe that is correct as of now. @candyho and their team have been working to add project functionality into the Mobile app but currently, you're limited to editing the project and project fields for a given issue as far as I know.

**Reply by mchelen-gov** [reply]
> I believe that is correct as of now. @candyho and their team have been working to add project functionality into the Mobile app but currently, you're limited to editing the project and project fields for a given issue as far as I know. 

Thanks for confirming! Is there any place to upvote this feature request? It would be wonderful to have full Project functionality on GH mobile.

**Reply by 2percentsilk** [reply]
Check out this [discussion](https://github.com/orgs/community/discussions/18127) :)

**Comment by henrzven** [comment]
Please fix GitHub Copilot it does not working on RubyMine.

**Comment by timbomckay** [comment]
## Question Regarding Repo Association

I'm trying to create a project that's strictly associated with a singular repo. In my GitHub Enterprise account I can create one in the Org but not directly in the repo. Similarly my personal account can only create one from the User profile. I just skimmed through the documentation and saw that this is the recommended approach for [creating a project](https://docs.github.com/en/issues/planning-and-tracking-with-projects/creating-projects/creating-a-project).

I thought I'd ask if it's the long term intent or if there's a plan to allow singular repo assignment? Seems very tedious to assign each item to a repo once it's created instead of it automatically assigned to the repo. Similarly, if we create an issue and have to assign it to the desired project.

Is there a recommended workflow to alleviate this tedious task? I can associate a repo with a project for "quick access", curious why a org could do something similar and reference a project within a repo for quick access?

> _Forgive me if this should be a separate discussion thread. I can extract if necessary._

**Reply by Harlalkhedar** [reply]
My account flagged please suggest me

**Reply by webfutureinc** [reply]
Was wondering the same thing, thanks for asking.

**Reply by prikkprikkprikk** [reply]
I second this! Even though I'm only working with one project at the moment, and it's the top one in my list of repos, it's annoying to have to select it every time.

**Reply by testman42** [reply]
I'm interested in this as well.

**Reply by 2percentsilk** [reply]
Thanks for the feedback - it sounds like allowing you to set a *default* repo for the project might be useful? This could avoid the scenario where you're having to select a repo each and every time you're upgrading a draft issue to a full fledged one.

The short answer is no, there isn't a way to associate a project with a singular repo in the way Projects (classic) used to work but we're still listening and evolving the projects offering based on your feedback so this is really appreciated! We understand this is a parity gap so we aim to address it but it not be a a direct copy of the former workflow. Rather I'm interested in the ways the current flow is tedious so we can evaluate holistically how we can best serve your needs.

**Comment by zer0Kerbal** [comment]
### This broke the CLI (gh).

`gh issue create ... --project "projectname"` 

_cannot_ assign an issue to the new project board (only classic) - _unless there is a way I do not know of_.

### ALSO I like classic projects.

Kindly allow us to choose if we use them or not.

I have many unrelated projects, each with it's own board; along with one supervisor user level project.

I like it this way.

Forcing us to use the 'new' projects only breaks much/many of my cli (gh/hub) scripts.

**Reply by timbomckay** [reply]
Curious if this has to do with my previous comment. Since projects are no longer created directly in a repo it might make it more difficult to link to a project via the terminal. Maybe if you've already linked it as "quick access" it might be able to locate it, but my experience has been you have to manually select the project from the dropdown.

But if you're saying this broke classic projects then yeah that's unfortunate.

**Reply by 2percentsilk** [reply]
Thanks for flagging this - we have it noted as an item to fix as part of our CLI parity with Projects (classics) but I don't have an ETA to share at this moment.

**Comment by BenjaminMyrstad** [comment]
Looks great. What Enterprise release will this be included in? 3.7?

**Comment by jdewinne** [comment]
Thanks. Is there a timeline for improving the GraphQL api and adding features like addMembersToProject, updateColumns to the api?
Looks like those were available in classic Projects, but not in ProjectV2?

**Reply by 2percentsilk** [reply]
Looking at this but no ETA to share at the moment. Safe to assume you're trying to automate creation of custom fields and adding items to your project? Are there other things in the API you're attempting that aren't working quite as expected?

**Comment by Gilbert1817** [comment]
How to access?

**Reply by 0814435859** [reply]
17

**Comment by HarshKapadia2** [comment]
I would love to be able to see the description of the issues/drafts in the board view.
Is there a way I can enable that?
Thank you!

**Reply by 2percentsilk** [reply]
Not at the moment but we're looking into how we can enable. Are you looking for a description that is separate from the body of the issue/draft or would rendering the body be sufficient for your needs?

**Comment by Ampmy999** [comment]
ไม่เป็นไร  ฉันกลัวทำผิด  ขอยังไม่ใส่ก็ได้

**Comment by melvincwng** [comment]
Awesome! Great to see such initiatives from the GitHub team! 

Cheers 🥂

**Comment by charlesreid1** [comment]
What happened to Projects Classic? What happened to the kanban board? That was so useful, but now it's been replaced with this ugly spreadsheet interface that forces me to organize around simple snippets of text and issues and PRs, instead of notes with rich Markdown. The Board view is difficult to find and switch to, and when you do, everything is issue-centric, there is no markdown support in the notes. Why would you get rid of such useful functionality? Why is the extremely useful Projects Classic being done away with and replaced with a less functional tool that is less flexible? I am not interested in this new Projects. Please make Projects Classic something that can be enabled. Thank you.

![Classic projects interface](https://docs.github.com/assets/cb-110288/images/help/projects/project-board-basic-kanban-template.png)

**Reply by timbomckay** [reply]
When you create a project you can create different views, one of them being a spreadsheet the alternative being a kanban board. Though I believe the cards only show titles and labels, not all the extra content you're displaying. Unless I haven't seen that setting yet.

**Reply by 2percentsilk** [reply]
@timbomckay is correct! However, your feedback on not having rich markdown for the card descriptions in the new Projects is heard. Would it suffice to have the body rendered as part of the cards or do you feel that a description attribute that is separate from the body is necessary for rendering in the board view?

**Reply by timbomckay** [reply]
I don't have an immediate need for the extra content as it'd make our boards significantly larger now that we have over 200 tickets in our board, but I know the flexibility would be appreciated by the community at large. Markdown would probably provide the most flexibility for stylized text, links, lists and even using checkboxes for subtasks as @charlesreid1's screenshot demonstrates.

**Comment by stevebrownlee** [comment]
I don't see any documentation in the Github API docs for creating new projects - only the classic projects. Is there any way to create a project via the API or am I forced to manually create them?

**Comment by jdewinne** [comment]
There is a way using the graphql API (ProjectsV2). But many things like configuring columns, adding users/members doesn't seem available (yet)?

**Reply by 2percentsilk** [reply]
Those are known gaps that we are hoping to close soon but no ETA to share as of yet.

**Comment by mrmetaverse** [comment]
Its not more enjoyable, its drastically changing the workflow. Why can't we create projects for repos anymore? It wants me to make my own personal project board and assign tickets to the repo. My team working on back end work needs a place to talk about back end work, so they have their own repo mycompany/backend with their own backlog "backend kanban" and I can track all teams from the org projects board (the new holistic one you have made). That worked for me. 

This new flow is so broken and so far from the last one that I am deeply concerned this wasn't tested with any real users.

**Reply by timbomckay** [reply]
Yeah this is similar to my [reply re:repo association](https://github.com/orgs/community/discussions/27327#discussioncomment-3310762). Though it sounds like this new structure would work well for your backend team to provide a kanban for the org (mycompany/projects/backend) instead of having to create a repo just for the kanban, but I might be misunderstanding your flow as it sounds like you're not in favor of that functionality.

I do agree with you though, the flow of it doesn't seem very intuitive and curious how they landed here and what the list of feature enhancements contain to restore some of the previous behavior.

**Reply by mrmetaverse** [reply]
I tried that out today, and it actually worked nicely! Unfortunately, we are still out of repo-specific milestones, and the ability to create issues from comments. I will keep exploring the new UX, but its extremely irresponsible to rip the band-aid off like this. They should incorporate a "Projects Classic" feature to continue using the old method (like I can on all of my old repos).

**Reply by timbomckay** [reply]
Ah yeah, you have to associate every new ticket with a repo and then you can access the labels & milestones from that repo. It's really weird and unintuitive. I do think there are a lot of improvements but the whole repo association bit is a big miss. Which would've made more sense to deprecate Classic when this new one was released instead of pulling the rug out.

The only benefit I see from this is for those organizations that have multiple repos associated with the same project. Like the Vue library, documentation, cli, etc are all separate repos and there's a core project that is able to assign tickets to all repos. That's my only guess and seems fairly abnormal.

Hopefully they improve this but it's been 10 days since my original post and haven't seen any responses of why they went in this direction.

**Comment by cIvanrc** [comment]
Awesome, it's so helpful.

Please, in the `Item closed` workflow, add support when an issue is closed as completed (merged) and when is just closed.
Something like this. 
![image](https://user-images.githubusercontent.com/29314633/184279798-4b9c4fcd-f40b-4cce-bcf9-81a9d423bdb6.png)

Also, please add support/event/workflow when an issue is linked/connected with a pull request.

Reading the previous comments I am seeing the flexibility that different people would give to Github projects, in my case I would like to focus it more on non-developers who would follow up on issues but who don't need to have a notion about pull requests. 

The scenario that I propose for my team is just to follow the issues (in the end, within the issue, it would have a pull request linked).

1.- When an issue is created automatically send it to the Backlog (which it already does)

2.- When an issue is connected to a pull request, update the status of the issue to 'In progress'

3.- When a pull request is merged, the issue is closed, and when the issue is closed the status should change to 'Merged' (this scenario already exists in current workflows, but currently it is not distinguished when an issue is closed by merged or closed as not planned)

**Comment by emin63** [comment]
The new projects feature is great! Can you provide a link or example for how to use the `gh` CLI to get a list of issues in a project (including title and issue number)? The closest example I could find was https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-api-to-manage-projects#finding-information-about-items-in-a-project but that is quite complicated.

A simple way to use the `gh` CLI to get a list of issues in a project would be great. Thanks.

**Comment by atirutw** [comment]
Wasn't there a plan for custom workflows or something? What happened to it?

**Reply by casc-bryl** [reply]
Same, I need to run some code when an issue is added or deleted on a board. Webhooks might not be possible in my use case.

**Comment by krisu5** [comment]
Bring back markdown cards! [My reasons](https://github.com/orgs/community/discussions/4422#discussioncomment-3415424):

> Without this feature, I will not begin using the new projects.
> It was the number one thing I used in classic projects (even with some flaws) and "drafts" won't cut it.
>In my view, projects and issues are seperate things. And in my repos, using projects for listing issues feels reduntant.
> I rather move to Notion or Trello then start using this. It is a shame, because then less likely people will know that's in the roadmap.

At least make it possible to see the content body and interact with it without opening dialogs.

**Comment by pookage** [comment]
Is there a way to migrate back to the 'classic' projects? I'm seeing a lot of missing automation features, and it's super frustrating not being able to navigate to the repo's discussions / wiki / code / etc pages when on the new project board...

**Reply by PennRobotics** [reply]
If you already have a repo with classic projects, you can _usually_ (but not always for some odd reason) copy the classic project (via the dropdown menu) into another repo under your ownership. This will then enable classic projects for the desired repo, and you can disable the new projects in the settings to completely transition back to classic projects.
